lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -2223,7 +2223,8 @@ sub should_show_autoinst_log {
 sub should_show_investigation {
     my ($self) = @_;
 
-    return $self->state ne DONE || $self->result eq FAILED;
+    OpenQA::Jobs::Constants::meta_state($self->state) eq OpenQA::Jobs::Constants::FINAL
+      && !OpenQA::Jobs::Constants::is_ok_result($self->result);
```

Comment:
This changes the behavior. I suppose

```suggestion
    OpenQA::Jobs::Constants::meta_state($self->state) ne OpenQA::Jobs::Constants::FINAL
      || !OpenQA::Jobs::Constants::is_ok_result($self->result);
```

would be correct.

Note that as long as the job state isn't final we need to render the tab server-side (even if it is hidden client-side) because when the job turns out to be a failure the tab is supposed to be shown after all (and thus set visible by the client-side after all).

---

Of course this means you cannot use this function in the way you're using it above. Maybe the function deserves a bitter name or the two slightly different usages should be inlined.

Reply:
Looks like I'm contradicting my previous comment (https://github.com/os-autoinst/openQA/pull/3981#discussion_r678298027) but I guess I just haven't thought it trough enough last time.

Reply:
Could you please test this?

Reply:
Confirmed. If a test is running and turns into failing the investigation tab is shown correctly (without reloading the page). That relies on the tab being rendered on the server-side for non-final states.

Reply:
By the way, I've tested that by simply setting the job state/result manually via SQL while having the tab open.

Reply:
Updated. I think the name of the method is still correct as is